src/phoenix/experimental/evals/models/rate_limiters.py

```diff
@@ -0,0 +1,230 @@
+        if self.rate != new_rate:
+            self.rate = new_rate
+            self.tokens = 0  # reset tokens as conservatively as possible
+            self.max_tokens = max_tokens
+            self.last_checked = time.time()
+
+    def available_tokens(self) -> float:
+        time_since_last_checked = time.time() - self.last_checked
+        return min(self.max_tokens, self.rate * time_since_last_checked + self.tokens)
+
+    def spend_tokens_if_available(self, token_cost: Numeric) -> None:
+        if token_cost > self.available_tokens():
+            raise UnavailableTokensError
+        now = time.time()
+        current_tokens = min(self.max_tokens, self.tokens + (now - self.last_checked) * self.rate)
+        self.tokens = current_tokens - token_cost
+        self.last_checked = now
+        self.total_tokens += token_cost
+
+    def spend_tokens(self, token_cost: Numeric) -> None:
+        self.tokens -= token_cost
+        self.total_tokens += token_cost
+
+    def wait_for_then_spend_available_tokens(
+        self, token_cost: Numeric, max_wait_time: Numeric = 300
+    ) -> None:
+        start = time.time()
+        while (time.time() - start) < max_wait_time:
+            try:
+                self.spend_tokens_if_available(token_cost)
+                break
+            except UnavailableTokensError:
+                time.sleep(1 / self.rate)
+                continue
+
+    async def async_wait_for_then_spend_available_tokens(
+        self, token_cost: Numeric, max_wait_time: Numeric = 300
+    ) -> None:
+        start = time.time()
+        while (time.time() - start) < max_wait_time:
+            try:
+                self.spend_tokens_if_available(token_cost)
+                break
+            except UnavailableTokensError:
+                await asyncio.sleep(1 / self.rate)
+                continue
+
+    def effective_rate(self) -> Numeric:
+        return self.total_tokens / (time.time() - self.created)
+
+
+class LimitStore:
+    """
+    A singleton store for collections of rate limits grouped by service key.
+
+    LimitStore is a singleton because all calls to a single endpoint should share the same rate
+    limits. Rate limiter implementations can use a LimitStore instance to manage rate limits for
+    a specific endpoint indexed by a key. The LimitStore singleton will ensure that rate limits
+    are correctly shared across all calls to the same endpoint.
+
+    This implementation is not threadsafe, but is async-safe.
+    """
+
+    _singleton = None
+
+    def __new__(cls) -> "LimitStore":
+        if not cls._singleton:
+            cls._singleton = super().__new__(cls)
+            cls._singleton._rate_limits = defaultdict(dict)
+        return cls._singleton
+
+    _rate_limits: Dict[str, Dict[str, TokenRateLimiter]]
+
+    def set_rate_limit(
+        self,
+        key: str,
+        limit_type: str,
+        per_minute_rate_limit: Numeric,
+        enforcement_window_minutes: Optional[int] = None,
```

Comment:
Can we give it a 1 to begin with, so the user can see it as part of the function signature?
```suggestion
        enforcement_window_minutes: int = 1,
```